configure.ac

```diff
@@ -1479,9 +1479,23 @@ if test "$use_boost" = "yes"; then
   BOOST_CPPFLAGS="$BOOST_CPPFLAGS -DBOOST_MULTI_INDEX_DISABLE_SERIALIZATION"
 
   dnl Prevent use of std::unary_function, which was removed in C++17,
-  dnl and will generate warnings with newer compilers.
-  dnl See: https://github.com/boostorg/container_hash/issues/22.
-  BOOST_CPPFLAGS="$BOOST_CPPFLAGS -DBOOST_NO_CXX98_FUNCTION_BASE"
+  dnl and will generate warnings with newer compilers for Boost
+  dnl older then 1.80.
+  dnl See:
+  dnl   - https://github.com/boostorg/container_hash/issues/22
```

Comment:
We only need to link to a single upstream issue, probably https://github.com/boostorg/config/pull/430.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/26952#issuecomment-1401808126).